cogs/music.py

```diff
@@ -19,154 +27,451 @@ def __init__(self, bot: ManChanBot):
+            return
+        id = int(" ".join(args).strip())
+        playlist = PlaylistDB.get_by_id(id)
+        if playlist is None:
+            await ctx.send(
+                f"Playlist: {' '.join(args).strip()} does not exist", delete_after=10
+            )
+            return
+
+        player = self.master_player.get_player(ctx.guild.id)
+        if player is None:
+            return
+        if player.current_song is None:
+            await ctx.send("No song is currently playing.", delete_after=10)
+            return
+
+        for song in player.queue:
+            addedSong = SongDB.get_by_url(song.url)
+            if addedSong is None:
+                addedSong = SongDB.create(song)
+            PlaylistSongDB.create(playlist.id, addedSong.id)
+        await ctx.send(f"Current Queue added to Playlist: {str(id)}", delete_after=10)
+
+    @command(aliases=["ahqpl"])
+    async def add_history_queue_playlist(self, ctx: Context, *args: str):
+        await ctx.message.delete(delay=5)
+        if ctx.guild is None:
+            await ctx.send("This command can only be used in a server.", delete_after=5)
+            return
+        id = int(" ".join(args).strip())
+        playlist = PlaylistDB.get_by_id(id)
+        if playlist is None:
+            await ctx.send(
+                f"Playlist: {' '.join(args).strip()} does not exist", delete_after=10
+            )
+            return
+
+        player = self.master_player.get_player(ctx.guild.id)
+        if player is None:
+            return
+        if player.current_song is None:
+            await ctx.send("No song is currently playing.", delete_after=10)
+            return
+
+        for song in player.history:
+            addedSong = SongDB.get_by_url(song.url)
+            if addedSong is None:
+                addedSong = SongDB.create(song)
+            PlaylistSongDB.create(playlist.id, addedSong.id)
+        await ctx.send(
+            f"Current history Queue added to Playlist: {str(id)}", delete_after=10
+        )
+
+    @command(aliases=["pls"])
+    async def list_playlist_songs(self, ctx: Context, *args: str):
+        await ctx.message.delete(delay=5)
+        id = int(" ".join(args).strip())
+        playlist = PlaylistSongDB.get_songs_by_playlist(id)
+        if playlist is None:
+            await ctx.send(
+                f"Playlist: {' '.join(args).strip()} does not exist", delete_after=10
+            )
+            return
+        embed = disnake.Embed()
+        playlist_name = PlaylistDB.get_by_id(id)
+        if playlist_name is None:
+            return
+        embed.title = playlist_name.playlist_name
+        playlist_str = ""
+        index = 1
+        for entry in playlist:
+            song = SongDB.get_by_id(entry.id)
+            if song is not None:
+                playlist_str += (
+                    str(index) + ". " + song.title + "(" + str(song.id) + ")" + "\n"
```

Comment:
you can simplify this with a f-string interpolation `f"{index}. {song.title} ({song.id})\n"

You can make it nicer by creating padding on the index to left align it. such that it looks like this:

```
 8.
 9.
10.
11.
```

https://www.geeksforgeeks.org/string-alignment-in-python-f-string/